### PR TITLE
fix:  update version to 2.4.1 docusaurus, use DocSearch v3 and use swaddle for SearchBar

### PR DESCRIPTION
**Pull request by isaac-shek**
#### 📲 What

- update version to 2.4.1 docusaurus
- use DocSearch v3
- use swaddle for SearchBar
- requires node 16.14 onwards when updating docusaurus to 2.4.1

#### 🤔 Why
currently implementation is using the docSearch old infrastructure, we will have to update the application and add 	"@docsearch/react" into package.json to get the new infrastructure.

#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence
		
Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR.
		 
#### 🕵️ How to test

Notes on how a reviewer can test the changes, e.g. how to run the tests.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -13,11 +13,6 @@ module.exports = {
           "developer/java/patterns/java_swagger_annotations",
         ],
       },
-      {
-        type: "category",
-        label: "Azure Development Patterns",
-        items: [],
```
Comment (isaac-shek) [inline sidebars.js:19]: Docusaurus does not allow empty items - this does not show in the docs, even on live

File: docusaurus.config.js [hunk 2]
```diff
@@ -45,15 +45,8 @@ module.exports = {
     ],
   },
   themeConfig: {
-    gtag: {
```
Comment (isaac-shek) [inline docusaurus.config.js:48]: bump up docusaurus, requires to move the gtag and googleAnalytics under presets

File: src/theme/SearchBar.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import '@docsearch/css';
+
+export default function SearchBarWrapper() {
+  return (
+    <>
+        {/* Can we put the appId, indexName and apiKey somewhere else? */}
+        <DocSearch
+          appId="HZJPMBSKK3"
+          indexName="stacks-amido"
+          apiKey="6b3dcd0dbf85334a8c18a2d62f1bff35"
```
Comment (leesaxby) [inline src/theme/SearchBar.js:13]: Could we setup some envars for these?

**Review by leesaxby — APPROVED** [review]

File: src/theme/SearchBar.js (hunk 3 above)
Comment (isaac-shek) [reply src/theme/SearchBar.js]: good shout, done

**Pull request merged**